### PR TITLE
User model refactor

### PR DESCRIPTION
**Pull request by mateen993**
Refactored user Model. Implemented pre-save middleware for hashing the password and While sending back the signed-up Data, removing the field 'password' and 'confirmPassword' field

### CODE REVIEW

**Comment by mateen993** [comment]
Kindly, see the changes

File: src/Model/user.model.mjs [hunk 1]
```diff
@@ -21,7 +48,11 @@ const UserSchema = new Schema(
         },
     }
 )
-
+UserSchema.pre('save', async function(next){
+    if(!this.isModified('password')) return next()
+    this.password = await bcrypt.hash(this.password,12)
+    this.confirmPassword = undefined
```
Comment (Ololawrence) [inline src/Model/user.model.mjs:54]: here you are returning confirm password to undefine why?

File: src/Model/user.model.mjs [hunk 2]
```diff
@@ -1,13 +1,40 @@
...
+            type: Boolean, 
+            default: false 
+        },
+        password: { 
+            type: String, 
+            required: [true, 'A user must have a password'],
+            minlength: [8, 'A password must have 8 or more characters'],
+            trim: true
+        },
+        confirmPassword: {
```
[the remark this answers is not in the document]
Comment (mateen993) [reply src/Model/user.model.mjs:27]: I created the logic in the user model as you can see under confirm password to check whether the password is equal to the confirmPassword or not! if not, throw an error..... This is just for validation....

File: src/Model/user.model.mjs (hunk 1 above)
Comment (mateen993) [reply src/Model/user.model.mjs:54]: I'm not returning anything... I'm undefining the confirm password field before saving it into the database, coz we don't want the confirm password field in the database, We only wanted this for checking if both passwords provided are equal. You can verify the behavior of code by checking different combos and analyzing the output. It seems alright to me

**Pull request merged**